apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/RetentionPolicyTest.cs

```diff
@@ -110,6 +110,64 @@ public void LockRetentionPolicy()
             Assert.True(bucket.RetentionPolicy.IsLocked);
         }
 
+        /// <summary>
+        /// Tests for object retention policy.
+        /// </summary>
+        [Fact]
+        public void ObjectRetentionTests()
+        {
+            var client = _fixture.Client;
+            var bucketName = _fixture.GenerateBucketName();
+
+            // Create a bucket with object retention enabled.
+            var createBucketoptions = new CreateBucketOptions { EnableObjectRetention = true };
+            var bucket = _fixture.CreateBucket(bucketName, false, createBucketoptions);
+            bucket = client.GetBucket(bucketName);
+            Assert.Equal("Enabled", bucket.ObjectRetention?.Mode);
+
+            var baseDateTime = DateTimeOffset.UtcNow;
+
+            string objectName = "object.txt";
+            // Create an object, with retention configured.\
+            var newObject = new Object
+            {
+                Bucket = bucketName,
+                Name = objectName,
+                ContentType = "text/plain",
+                Retention = new Object.RetentionData
+                {
+                    Mode = "Unlocked",
+                    RetainUntilTimeDateTimeOffset = baseDateTime.AddDays(5),
+                }
+
+            };
+            CreateObject(newObject);
+            var obj = client.GetObject(bucketName, objectName);
+            Assert.Equal("Unlocked", obj.Retention?.Mode);
+            Assert.Equal(baseDateTime.AddDays(5).Date, obj.Retention.RetainUntilTimeDateTimeOffset.Value.Date);
```

Comment:
Or am I missing something?

```suggestion
            Assert.Equal(baseDateTime.AddDays(5), obj.Retention.RetainUntilTimeDateTimeOffset.Value);
```

Reply:
I would suggest truncating `baseDateTime` to the millisecond to start with (with a comment), then performing an exact comparison.